### ISSUE TITLE
magicsock: move stun servers to 41641

### ISSUE THREAD

**Issue by crawshaw** [tailscale/tailscale#2574]
If you wanted to engaged in the (discouraged) practice of manually writing allow rules for every piece of outgoing traffic Tailscale makes, you would need to include:

```
TCP controlplane.tailscale.com:443
TCP login.tailscale.com:443
UDP *:41641
TCP derp*.tailscale.com:443
UDP derp*.tailscale.com:3478
```

The last rule is just for STUN packets. We could simplify these by sending our stun packets to `derp*.tailscale.com:41641`. (Which would also be a useful target for any future DERP-over-UDP protocol we develop.)

**Comment by bradfitz** [comment]
Related: https://github.com/tailscale/tailscale/issues/1872 (Publish Little Snitch ruleset for Tailscale)

In general I remain nervous about going down the path of publishing rules saying "only UDP XXX outgoing is needed!" because portmapping means clients will end up with all sorts of random UDP ports and if we get a bunch of people blocking non-41641 UDP out, that means direct connections will fail and everybody will pay the DERP cost (us in money, users in reduced speed).

I'd rather our rules say "any UDP outgoing" and instead focus on incoming: "41641 UDP in". From the host machine's perspective, only 41641 arrives in, even if their router doing the portmapping has a different public port number.

**Comment by crawshaw** [comment]
I agree in general we should be encouraging allowing UDP out to all ports and destinations.

**Comment by bradfitz** [comment]
> I agree in general we should be encouraging allowing UDP out to all ports and destinations.

So the sole purpose of this bug/proposal is to make things easier for people doing the things we don't want them to do? :)
